core/src/main/java/org/apache/accumulo/core/file/blockfile/cache/lru/CachedBlockQueue.java

```diff
@@ -91,24 +93,19 @@ public void add(CachedBlock cb) {
    * @return list of cached elements in descending order
    */
   public CachedBlock[] get() {
-    LinkedList<CachedBlock> blocks = new LinkedList<>();
-    while (!queue.isEmpty()) {
-      blocks.addFirst(queue.poll());
-    }
-    return blocks.toArray(new CachedBlock[blocks.size()]);
+     CachedBlock[] blocks = queue.toArray(new CachedBlock[0]);
+     Arrays.sort(blocks, Collections.reverseOrder(queue.comparator()));
+     queue.clear();
+     return blocks;
   }
 
   /**
    * Get a sorted List of all elements in this queue, in descending order.
    *
    * @return list of cached elements in descending order
    */
-  public LinkedList<CachedBlock> getList() {
-    LinkedList<CachedBlock> blocks = new LinkedList<>();
-    while (!queue.isEmpty()) {
-      blocks.addFirst(queue.poll());
-    }
-    return blocks;
+  public List<CachedBlock> getList() {
+    return Arrays.asList(get());
```

Comment:
This could certainly be consolidated with the code in `get()`, but you could also achieve that by leaving this method as-is, and having the the `get()` method's implementation call `getList()`.

Reply:
Ya, I was trying to figure out the best way to do this.

Doing this way creates only a single array in the `get()` method.   The `Arrays.asList()` `Returns a fixed-size list backed by the specified array.`

https://docs.oracle.com/javase/7/docs/api/java/util/Arrays.html#asList(T...)

Doing it the other way, I've got to create the ArrayList (1 array) and then call `.toArray()` which creates a second copy of the array.  Besides, I'm pretty sure nothing internal to the project called `getList()` so it's better to optimize for `get()`

Reply:
Using `Array.AsList()` also makes the two methods a bit more symmetrical.  One cannot add a new element to the List returned by `getList()` just as one cannot add an item to the native array type returned by `get()`

Reply:
If we don't use `getList()` in the project, it should just be removed, and optimize `get()` as you suggest. This isn't public API, so we're free to change the internals without breaking any user contract.

Reply:
I believe that `getList()` was only used as a wrapper for unit tests.  I have removed the method but waiting on CI to check that my IDE wasn't lying about usage. :)

Reply:
Removing it was a nice change. Thanks! :smiley_cat: